src/analyze/analyze-security.c

```diff
@@ -1798,11 +1806,11 @@ static int assess(const SecurityInfo *info,
                         if (d)
                                 description = d;
 
-                        if (json_variant_by_key(policy, a->json_field) != NULL)
+                        if (json_variant_by_key(policy, a->json_field) != NULL || json_format_flags != JSON_FORMAT_OFF)
                                 id = a->json_field;
```

Comment:
actually i think we should always add both `a->json_field` *and* `a->id` to the table, in separate columns, and include both in the JSON output, but hide the json field name from the regular text output, using `table_set_display()`.

i.e. it's a good thing if the table is always filled the same way, and we simply tweak what parts we show of it after it is put together depending on JSON or not.

Reply:
How does table_set_display() work because it seems like adding these lines to the code is just producing an 8 column table with columns: " ", id, json_field, description, exposure, " ", id, description, exposure.

    if (json_format_flags & JSON_FORMAT_OFF)
                        r = table_set_display(details_table, (size_t) 0, (size_t) 1, (size_t) 3, (size_t) 7);
                else
                        r = table_set_display(details_table, (size_t) 0, (size_t) 1, (size_t) 2, (size_t) 3, (size_t) 7);
                if (r < 0)
                        return log_error_errno(r, "Failed to set columns to display: %m");

And the json_field seems to be included for both the off and non off cases of JSON. What am I missing?